beets/dbcore/query.py

```diff
@@ -107,9 +112,9 @@ def __hash__(self) -> int:
         return hash(type(self))
 
 
-P = TypeVar("P")
-SQLiteType = Union[str, bytes, float, int, memoryview]
+SQLiteType = Union[str, bytes, float, int, memoryview, None]
 AnySQLiteType = TypeVar("AnySQLiteType", bound=SQLiteType)
+FieldQueryType = type["FieldQuery"]
```

Comment:
Personally, I'd be in favor of sticking with the explicit `type[FieldQuery]`, it seems more obvious to me what it is supposed to be (note how `Type` means different things in `AnySQLiteType` and `FieldQueryType`).

In any case, not a strong opinion here: Feel free to stick to your preference!

Reply:
One of the main motivations I had for this was the presence of `types.Type`: in some places it was confusing to see `types.Type` next to `type`, so this helped to remove some of the ambiguity.

Reply:
Ok! Probably, whatever we do, there will some ambiguity around the term "type" in `dbcore`.